src/providers/wfs/qgswfsshareddata.cpp

```diff
@@ -93,6 +93,43 @@ QString QgsWFSSharedData::srsName() const
   return srsName;
 }
 
+QString QgsWFSSharedData::computedExpression( const QgsExpression &expression ) const
+{
+  QgsOgcUtils::GMLVersion gmlVersion;
+  QgsOgcUtils::FilterVersion filterVersion;
+  bool honourAxisOrientation = false;
+  if ( mWFSVersion.startsWith( QLatin1String( "1.0" ) ) )
+  {
+    gmlVersion = QgsOgcUtils::GML_2_1_2;
+    filterVersion = QgsOgcUtils::FILTER_OGC_1_0;
+  }
+  else if ( mWFSVersion.startsWith( QLatin1String( "1.1" ) ) )
+  {
+    honourAxisOrientation = !mURI.ignoreAxisOrientation();
+    gmlVersion = QgsOgcUtils::GML_3_1_0;
+    filterVersion = QgsOgcUtils::FILTER_OGC_1_1;
+  }
+  else
+  {
+    honourAxisOrientation = !mURI.ignoreAxisOrientation();
+    gmlVersion = QgsOgcUtils::GML_3_2_1;
+    filterVersion = QgsOgcUtils::FILTER_FES_2_0;
+  }
```

Comment:
those lines, copied&pasted from QgsWFSSharedData::computeFilter(), could be moved in a small helper function